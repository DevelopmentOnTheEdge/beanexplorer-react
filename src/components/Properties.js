import React from 'react';
import PropTypes from 'prop-types';
import Property from './Property';
import JsonPointer from 'json-pointer';

class Properties extends React.Component {

  shouldComponentUpdate(nextProps) {
    return this.props.bean !== nextProps.bean || this.props.values !== nextProps.values;
  }

  render() {
    let fields = this.props.bean.order.map((path, i) => {
      if (this.props.ids === undefined || this.props.ids.includes(i)) {
        return (
          <Property path={path} key={path} {...this.props} value={this.getValue(path)}/>
        )
      } else {
        return null;
      }
    });

    //todo remove outer element after migrate to react 16.2
    return <div className={this.props.rowClass}>{fields}</div>;
  }

  getValue(path) {
    const values = this.props.values || this.props.bean.values;
    return JsonPointer.get(values, path)
  }
}

Properties.defaultProps = {
  rowClass: "form-row",
};

Properties.propTypes = {
  rowClass: PropTypes.string,
  bean: PropTypes.object.isRequired,
  ids: PropTypes.array,
  inline: PropTypes.bool,
  bsSize: PropTypes.string,
  onChange: PropTypes.func,
  reloadOnChange: PropTypes.func,
  localization: PropTypes.object
};

export default Properties;
