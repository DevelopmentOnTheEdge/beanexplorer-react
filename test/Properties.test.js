import React from 'react';
import Properties from '../src/components/Properties';
import renderer from 'react-test-renderer';
import bean from '../src/testJson.json';


test('contain property', () => {
	const component = renderer.create(
    <Properties bean={bean} ids={[0,15]}/>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('not contain property', () => {
  const component = renderer.create(
    <Properties bean={bean} ids={[1000000]}/>
  );
  expect(component.toJSON()).toMatchSnapshot();
});


