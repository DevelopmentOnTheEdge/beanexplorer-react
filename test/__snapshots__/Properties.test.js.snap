// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contain property 1`] = `
<div
  className="form-row"
>
  <div
    className="vertical-input col-lg-12"
  >
    <div
      className="property form-group required"
    >
      <label
        className="form-control-label"
        htmlFor="textInputPropertyInput"
      >
        Text input
      </label>
      <div
        className="controls"
      >
        <input
          className="property-input form-control"
          id="textInputPropertyInput"
          maxLength={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onInput={[Function]}
          onInvalid={[Function]}
          pattern={undefined}
          placeholder={undefined}
          required={true}
          size={undefined}
          type="text"
          value="test"
        />
      </div>
    </div>
  </div>
  <div
    className="vertical-input col-lg-12"
  >
    <div
      className="property-label property form-group required"
    >
      <label
        className="property-input"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Pellentesque habitant morbi tristique senectus et netus et <a href=\\"#test\\">Link</a> malesuada fames ac <b>turpis</b> egestas.</p>",
          }
        }
        id="labelFieldRawPropertyInput"
      />
    </div>
  </div>
</div>
`;

exports[`not contain property 1`] = `
<div
  className="form-row"
/>
`;
